libraries/core_libs/consensus/src/pillar_chain/pillar_block.cpp

```diff
@@ -57,11 +62,59 @@ PillarBlock::Hash PillarBlock::getHash() {
   }
 
   std::scoped_lock lock(hash_mutex_);
-  hash_ = dev::sha3(getRlp());
+  hash_ = dev::sha3(encodeSolidity());
   return *hash_;
 }
 
-RLP_FIELDS_DEFINE(PillarBlock, period_, state_root_, previous_pillar_block_hash_, validators_stakes_changes_)
+Json::Value PillarBlock::getJson() const {
+  Json::Value res;
+  res["period"] = dev::toJS(period_);
+  res["state_root"] = dev::toJS(state_root_);
+  res["bridge_root"] = dev::toJS(bridge_root_);
+  res["previous_pillar_block_hash"] = dev::toJS(previous_pillar_block_hash_);
+  res["validators_stakes_changes"] = Json::Value(Json::arrayValue);
+  for (auto const& stake_change : validators_stakes_changes_) {
+    Json::Value stake_change_json;
+    stake_change_json["address"] = dev::toJS(stake_change.addr_);
+    stake_change_json["value"] = dev::toJS(stake_change.stake_change_);
+
+    res["validators_stakes_changes"].append(std::move(stake_change_json));
+  }
+  res["hash"] = dev::toJS(getHash());
+
+  return res;
+}
+
+dev::bytes PillarBlock::encodeSolidity() const {
+  dev::bytes result;
+  result.reserve((1 + 4 + 2 + 2 * validators_stakes_changes_.size()) * util::EncodingSolidity::kWordSize);
```

Comment:
@kstdl what do these hardcoded values represent ?

Reply:
1. So 1 is the start, 4 is count of variables. 2 should be for position and size of the dynamic data. It is better to move it ot separate variables to be cleaner 
2. Maybe I was doing something wrong there. But `stake change(s256) + address` is `2 * validators_stakes_changes_.size()`(all values are 32 bytes)